pandas/io/formats/style.py

```diff
@@ -1060,12 +1061,76 @@ def hide_columns(self, subset) -> Styler:
         -------
         self : Styler
         """
-        subset = non_reducing_slice(subset)
-        hidden_df = self.data.loc[subset]
-        hcols = self.columns.get_indexer_for(hidden_df.columns)
-        # error: Incompatible types in assignment (expression has type
-        # "ndarray", variable has type "Sequence[int]")
-        self.hidden_columns = hcols  # type: ignore[assignment]
+        return self.hide_values(subset)
+
+    def hide_values(self, subset, axis: Axis = "columns", show: bool = False) -> Styler:
```

Comment:
i find this a very confusing name, hide_axis is more appropriate

Reply:
The trick is to distinguish between the cases of:

 a) hiding select rows or columns of data (whilst other index keys or column headers are visible),
 b) or displaying the data values but just hiding the index or column headers row in their entirety.

Currently `hide_columns` does a) whilst `hide_index` does b). The complete idea was that:

 i) `hide_values` would do a) for either the index or columns axes.
ii) `hide_headers` (a new method) would do b) for either the index or columns axes.

?

Reply:
@jreback i created an alternative in #41266 which seeks to achieve the same functionality in a different way re-using exiting methods and kwargs. Close this PR if you prefer the alternative,